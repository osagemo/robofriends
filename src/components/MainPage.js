import React, { Component } from "react";
import ErrorBoundry from "../components/ErrorBoundry";
import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";
import Header from "../components/Header";
import SimpleBar from "simplebar-react";

import "./MainPage.css";
import "simplebar/dist/simplebar.min.css";

export default class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () =>
    this.props.robots.filter(robot =>
      robot.name.toLowerCase().includes(this.props.searchField)
    );

  render() {
    const { onSearchChange, robots, isPending } = this.props;
    const filteredRobots = this.filterRobots(robots);

    if (isPending) {
      return <h1>Fetching users...</h1>;
    }

    return (
      <div className="tc">
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <SimpleBar style={{ maxHeight: "75vh" }}>
          <ErrorBoundry>
            <CardList robots={filteredRobots} />
          </ErrorBoundry>
        </SimpleBar>
      </div>
    );
  }
}
