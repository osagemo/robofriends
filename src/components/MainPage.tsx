import React, { Component } from "react";
import ErrorBoundary from "./ErrorBoundary";
import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";
import Header from "../components/Header";
import SimpleBar from "simplebar-react";

import "./MainPage.css";
import "simplebar/dist/simplebar.min.css";

export interface IRobot {
  name: string;
  id: number;
  email: string;
}

interface IMainPageProps {}

interface IMainPageState {
  robots: Array<IRobot>;
  searchField: string;
}

export default class MainPage extends Component<
  IMainPageProps,
  IMainPageState
> {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () =>
    this.props.robots.filter(robot =>
      robot.name.toLowerCase().includes(this.props.searchField)
    );

  render() {
    const { onSearchChange, robots, isPending } = this.props;
    const filteredRobots = this.filterRobots(robots);

    if (isPending) {
      return <h1>Fetching users...</h1>;
    }

    return (
      <div className="tc">
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <SimpleBar style={{ maxHeight: "75vh" }}>
          <ErrorBoundary>
            <CardList robots={filteredRobots} />
          </ErrorBoundary>
        </SimpleBar>
      </div>
    );
  }
}
