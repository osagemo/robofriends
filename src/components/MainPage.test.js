import React from "react";
import { shallow } from "enzyme";
import MainPage from "./MainPage";

let wrapper;
let mockProps;

beforeEach(() => {
  mockProps = {
    robots: [],
    searchField: "",
    isPending: false,
    onRequestRobots: jest.fn(),
    onSearchChange: jest.fn()
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it("expect to render MainPage", () => {
  expect(wrapper).toMatchSnapshot();
});

it("can filter the robots state", () => {
  mockProps = {
    robots: [
      {
        id: 1,
        name: "Kunta",
        email: "king@kunta.rymz"
      },
      {
        id: 2,
        name: "Busta",
        email: "Bus@ta.rymz"
      }
    ],
    searchField: "k",
    isPending: false,
    onRequestRobots: jest.fn(),
    onSearchChange: jest.fn()
  };
  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper.instance().filterRobots()).toEqual([
    {
      id: 1,
      name: "Kunta",
      email: "king@kunta.rymz"
    }
  ]);
});

it("shows a message when pending", () => {
  mockProps.isPending = true;
  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper.instance().render()).toEqual(<h1>Fetching users...</h1>);
});
