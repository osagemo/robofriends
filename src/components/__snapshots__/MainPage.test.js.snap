// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox
    searchChange={[MockFunction]}
  />
  <SimpleBar
    style={
      Object {
        "maxHeight": "75vh",
      }
    }
  >
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </SimpleBar>
</div>
`;
